File: 2015/Day 1/main.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Day1 {
    public static void main(String[] args) {
        int answer1 = part1();
        System.out.printf("Answer to part 1: %d\n",answer1);
        int answer2 = part2();
        System.out.printf("Answer to part 2: %d\n",answer2);
    }

    public static int part1() {
        try {
            File file = new File("Day1Input.txt");
            Scanner fileReader = new Scanner(file);
            fileReader.useDelimiter("");
            int floor = 0;
            while(fileReader.hasNext()) {
                String inst = fileReader.next();
                if(inst.compareTo("(") == 0) floor++;
                else floor--;
            }
            fileReader.close();
            return floor;
        }
        catch (FileNotFoundException e) {
            System.out.println("An error occurred");
            e.printStackTrace();
            return -1;
        }
    }

    public static int part2() {
        try {
            File file = new File("Day1Input.txt");
            Scanner fileReader = new Scanner(file);
            fileReader.useDelimiter("");
            int floor = 0, i = 0;
            while(fileReader.hasNext()) {
                String inst = fileReader.next();
                if(inst.compareTo("(") == 0) floor++;
                else floor--;
                i++;
                if(floor == -1) break;
            }
            fileReader.close();
            return i;
        }
        catch (FileNotFoundException e) {
            System.out.println("An error occurred");
            e.printStackTrace();
            return -1;
        }
    }
}

File: 2015/Day 1/main.c
#include <stdio.h>

int part1() {
    FILE *file = fopen("Day1Input.txt","r");
    int final_floor = 0;
    char inst;
    while((inst = fgetc(file)) != EOF) {
        if(inst == '(') final_floor++;
        else final_floor--;
    }
    return final_floor;
}

int part2() {
    FILE *file = fopen("Day1Input.txt","r");
    int floor = 0, i = 0;
    char inst;
    while((inst = fgetc(file)) != EOF) {
        if(inst == '(') floor++;
        else floor--;
        i++;
        if(floor == -1) return i;
    }
}

int main(int argc, char const *argv[]) {
    int answer1 = part1();
    printf("Answer to part 1: %d\n",answer1);
    int answer2 = part2();
    printf("Answer to part 2: %d\n",answer2);
    return 0;
}
